doc/sphinx/proof-engine/ltac.rst

```diff
@@ -844,66 +1040,126 @@ We can carry out pattern matching on terms with:
-      rather than trying a new matching if the right-hand side fails. If
-      the right-hand side of the selected branch is a tactic with
-      backtracking points, then subsequent failures cause this tactic to
-      backtrack.
+         Ltac f t := match t with
+                    | context [ (not ?t) ] => idtac "?t = " t; fail
+                    | _ => idtac
+                    end.
+         Goal True.
 
-   .. tacv:: context @ident [@cpattern]
+      .. coqtop:: all
 
-      This special form of patterns matches any term with a subterm matching
-      cpattern. If there is a match, the optional :n:`@ident` is assigned the "matched
-      context", i.e. the initial term where the matched subterm is replaced by a
-      hole. The example below will show how to use such term contexts.
+         f ((not True) <> (not False)).
 
-      If the evaluation of the right-hand-side of a valid match fails, the next
-      matching subterm is tried. If no further subterm matches, the next clause
-      is tried. Matching subterms are considered top-bottom and from left to
-      right (with respect to the raw printing obtained by setting the
-      :flag:`Printing All` flag).
+.. _match_with_holes_ex:
 
-   .. example::
+   .. example:: Matching a pattern with holes
 
-      .. coqtop:: all abort
+      Notice the :tacn:`idtac` prints ``(z + 1)`` while the :tacn:`pose` substitutes
+      ``(x + 1)``.
+
+      .. coqtop:: in reset
 
-         Ltac f x :=
-           match x with
-             context f [S ?X] =>
-             idtac X;                    (* To display the evaluation order *)
-             assert (p := eq_refl 1 : X=1);    (* To filter the case X=1 *)
-             let x:= context f[O] in assert (x=O) (* To observe the context *)
+         Goal True.
+
+      .. coqtop:: all
+
+           match constr:(fun x => (x + 1) * 3) with
+           | fun z => ?y * 3 => idtac "y =" y; pose (fun z: nat => y * 5)
            end.
+
+.. _match_vs_lazymatch_ex:
+
+   .. example:: Comparison of match, multimatch and lazymatch
+
+      .. todo PR: how do match and multimatch differ?**
+
+      .. coqtop:: reset in
+
+         Goal True.
+
+      .. coqtop:: all
+
+         match False with
+         | False => idtac "try"; fail
+         | _ => easy
+         end.
+
+      .. coqtop:: reset in
+
+         Goal True.
+
+      .. coqtop:: all
+
+         multimatch False with
+         | False => idtac "try"; fail
+         | _ => easy
+         end.
+
+      .. coqtop:: reset in
+
          Goal True.
-         f (3+4).
+
+      .. coqtop:: all fail
+
+         lazymatch False with
+         | False => idtac "try"; fail
+         | _ => easy
+         end.
 
 .. _ltac-match-goal:
 
-Pattern matching on goals
-~~~~~~~~~~~~~~~~~~~~~~~~~
+Pattern matching on goals: match goal
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. todo PR The title seems incomplete, should be "goals and hypotheses", right?
```

Comment:
Yes. Sometimes people say simply "goal" to mean the actual "goal" to prove and its goal context.